lib/seahorn/Smt/TypeChecker.cc

```diff
@@ -62,8 +146,8 @@ class TypeChecker::Impl {
 public:
   Expr typeOf(Expr e) {
     TCV _visitor;
-    visit(_visitor, e);
-    return _visitor.knownTypeOf(e);
+    Expr v = dagVisit(_visitor, e);
```

Comment:
probably `visit` is just as good since your type-checker maintains the cache already. 
This way everything is cached twice.